File: tide-canvas-go/internal/model/canvas.go
package model

// CanvasProject 画布项目表 canvas_project。
type CanvasProject struct {
	PublicModel
	UserID      int64  `json:"-" gorm:"column:user_id"`
	Name        string `json:"name" gorm:"column:name"`
	Description string `json:"description" gorm:"column:description"`
	Thumbnail   string `json:"thumbnail" gorm:"column:thumbnail"`
	// CanvasData 画布 JSON（LONGTEXT）；列表查询应不 SELECT 此列，仅详情返回。
	CanvasData string `json:"canvasData,omitempty" gorm:"column:canvas_data"`
	IsPublic   int    `json:"isPublic" gorm:"column:is_public"`
	ShareToken string `json:"shareToken,omitempty" gorm:"column:share_token"`
	URLToken   string `json:"urlToken,omitempty" gorm:"column:url_token"`
	Status     int    `json:"status" gorm:"column:status"`
}

// TableName 表名。
func (CanvasProject) TableName() string { return "canvas_project" }

File: tide-canvas-web/src/types/canvas.ts
import type { PageQuery } from "./api";
import type { UserSimpleVO } from "./user";

export interface ProjectVO {
  id: string; // 后端雪花 ID 序列化为字符串(flowingLight 全站 string ID,勿改回 number)
  /** 归属用户ID（团队共享时区分自己/队友的项目） */
  ownerId?: string;
  name: string;
  description: string;
  thumbnail: string;
  status: ProjectStatus;
  isPublic: boolean;
  urlToken: string;
  /** Whole-canvas optimistic-lock version; metadata edits do not change it. */
  revision: number;
  createTime: string;
  updateTime: string;
}

export interface ProjectDetailVO extends ProjectVO {
  canvasData: string;
  shareToken: string;
  owner: UserSimpleVO;
}

export interface CanvasDataVO {
  canvasData: string;
  revision: number;
}

export interface CanvasSaveVO {
  revision: number;
}

export interface ShareVO {
  shareToken: string;
  shareUrl: string;
}

export interface ProjectCreateDTO {
  name: string;
  description?: string;
  /** 客户端稳定请求号；响应丢失时以同一值重放，避免重复创建项目。 */
  clientRequestId?: string;
}

export interface ProjectUpdateDTO {
  name?: string;
  description?: string;
  status?: ProjectStatus;
  isPublic?: boolean;
}

export interface CanvasSaveDTO {
  canvasData: string;
  thumbnail?: string;
  expectedRevision: number;
}

export interface ProjectQuery extends PageQuery {
  keyword?: string;
  status?: ProjectStatus;
}

export enum ProjectStatus {
  DRAFT = 0,
  PUBLISHED = 1,
}
